auth/src/main/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivity.java

```diff
@@ -229,14 +228,32 @@ private void populateIdpListCustomLayout(List<IdpConfig> providerConfigs) {
         for (IdpConfig idpConfig : providerConfigs) {
             final String providerId = providerOrEmailLinkProvider(idpConfig.getProviderId());
 
-            if (!providerButtonIds.containsKey(providerId)) {
+            Integer buttonResId = providerButtonIds.get(providerId);
+            if (buttonResId == null) {
                 throw new IllegalStateException("No button found for auth provider: " + idpConfig.getProviderId());
             }
 
-            @IdRes int buttonId = providerButtonIds.get(providerId);
+            @IdRes int buttonId = buttonResId;
             View loginButton = findViewById(buttonId);
             handleSignInOperation(idpConfig, loginButton);
         }
+        //hide custom layout buttons that don't have their identity provider set
+        for (String providerBtnId : providerButtonIds.keySet()) {
+            if (providerBtnId == null) continue;
+            boolean hasProvider = false;
+            for (IdpConfig idpConfig : providerConfigs) {
+                if (providerBtnId.equals(idpConfig.getProviderId())) {
+                    hasProvider = true;
+                    break;
+                }
+            }
+            if (!hasProvider) {
+                Integer resId = providerButtonIds.get(providerBtnId);
+                if (resId == null) continue;
```

Comment:
Is this null condition even possible?  How would we get into this state?

Reply:
Well, it's an unboxing, In theory I know that map should only have provider ids and layout res ids. But since that Map is passed around by reference, placing an Integer in it that's later nullified, you can end up with a null here. Better safe than sorry.